camp_fin/management/commands/import_transactions.py

```diff
@@ -129,7 +210,12 @@ def import_expenditures(self, f, year):
                         received_date__year=year,
                     ).delete()
 
-                self.make_contribution(record, None, filing).save()
+                contribution = self.make_contribution(record, None, filing)
+                batch.append(contribution)
+
+                if not len(batch) % batch_size:
+                    self._save_batch(batch)
+                    batch = []
```

Comment:
you also need to handle the case that you have iterated through all the records, but the batch  isn't modulo the batch_size

Reply:
Great catch, thanks.